machines/vic/install.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/machines/vic/config"
+	"github.com/vmware/vic/machines/vic/reification"
+
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+const (
+	// Max permitted length of Virtual Machine name
+	MaxVirtualMachineNameLen = 80
+	// Max permitted length of Virtual Switch name
+	MaxDisplayNameLen = 31
+)
+
+// Data is pulled from govmomi tests where we needed a global config as we could pass it around
+type Data struct {
+	target          string
+	user            string
+	passwd          string
+	computeResource string
+	imageDatastore  string
+	displayName     string
+
+	containerDatastore string
+	externalNetwork    string
+	managementNetwork  string
+	bridgeNetwork      string
```

Comment:
Naming here is kind of misleading. They are not the objects but their names, so how about naming them  as XyZName